test/behaviour/graql/TheoremSteps.java

```diff
@@ -0,0 +1,138 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package grakn.client.test.behaviour.graql;
+
+import com.google.common.collect.Iterators;
+import grakn.client.GraknClient;
+import grakn.client.answer.ConceptMap;
+import grakn.client.test.behaviour.connection.ConnectionSteps;
+import graql.lang.Graql;
+import graql.lang.query.GraqlGet;
+import graql.lang.query.GraqlInsert;
+import graql.lang.query.GraqlQuery;
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+public class TheoremSteps {
+
+    private static List<ConceptMap> answers;
+    private static RefreshingTransaction refreshingTx;
+
+    static class RefreshingTransaction {
+        private GraknClient.Transaction maybeNullTransaction = null;
+        GraknClient.Transaction getTransaction() {
+            if (maybeNullTransaction == null || maybeNullTransaction.isClosed()) {
+                GraknClient.Session session = Iterators.getOnlyElement(ConnectionSteps.sessions.iterator());
+                maybeNullTransaction = session.transaction().write();
+            }
+            return maybeNullTransaction;
+        }
+
+        void close() {
+            if (maybeNullTransaction != null) {
+                maybeNullTransaction.close();
+            }
+        }
+    }
+
+    private GraknClient.Transaction tx() {
+        return refreshingTx.getTransaction();
+    }
+
+    @Before
+    public void create_refreshing_tx() {
+        refreshingTx = new RefreshingTransaction();
+
+    }
+
+    @After
+    public void close_transaction() {
+        refreshingTx.close();
+    }
+
+    @Given("the KB is valid")
+    public void check_kb_is_valid() {
+
+    }
+
+    @Given("the schema")
+    public void graql_define_schema(List<String> defineQueryStatements) {
+        GraqlQuery graqlQuery = Graql.parse(String.join("\n", defineQueryStatements));
+        tx().execute(graqlQuery);
+        tx().commit();
+    }
+
+    @Given("the data")
+    public void graql_insert_data(List<String> insertQueryStatements) {
+        GraqlQuery graqlQuery = Graql.parse(String.join("\n", insertQueryStatements));
+        tx().execute(graqlQuery);
+        tx().commit();
+    }
+
+    @When("executing")
+    public void graql_query(List<String> graqlQueryStatements) {
```

Comment:
The best practice in defining Cucumber steps is to have the method name to be as identical as possible to the step syntax - i.e. by simply turning the step syntax into the snake_case version of them. I actually agree to this practice.
1) It makes it easier to debug when errors are thrown (when method and step names are the same)
2) Snake case is better than camelCase when you have very long names (in which tests do)

So let's stick to the above practice.

However, in addition to the above, notice how in the test name you came up with 3 consistent names that makes up the test scenarios? `graql_define`, `graql_insert` and `graql_query`. Let's also use this style for the step definitions in the feature files!